src/datasets/load.py

```diff
@@ -1507,16 +1642,27 @@ def load_dataset_builder(
         data_dir=data_dir,
         data_files=data_files,
     )
-
+    dataset_name = (
+        camelcase_to_snakecase(os.path.basename(path))
+        if dataset_module.module_path.split(".")[-1] in _PACKAGED_DATASETS_MODULES
+        else None
+    )
```

Comment:
An idea/suggestion: maybe the `dataset_name` could be defined inside `PackagedDatasetModuleFactory.get_module`, `LocalDatasetModuleFactoryWithoutScript.get_module` and `HubDatasetModuleFactoryWithoutScript.get_module`... and returned inside `builder_kwargs`.

Reply:
That way there would be separation of concerns: no logic here to find out which module factory got the module; instead, each module factory is responsible of generating their corresponding `dataset_name` if necessary (as default is None in `DatasetBuilder.__init__`.